src/loader-new.js

```diff
@@ -0,0 +1,342 @@
+
+    return;
+  }
+
+  /**
+   * Add a gray default placeholder if there isn't a placeholder already and
+   * other special cases.
+   * @private
+   */
+  maybeAddDefaultPlaceholder_() {
+    // NOTE(aghassemi): I do believe we need to exclude amp-list here, but
+    // let's see how experimentation goes. Maybe a better idea is to have a
+    // white list, any component that does not fully load a new background may
+    // look bad if a gray placeholder shows up and goes away quickly. This
+    // default placeholder is good for image, video, etc.. but amp-list which
+    // usually just loads text is debatable. amp-iframe is also a candidate
+    // to exclude, often it does load a video or maps but it may not load
+    // text on transparent background in certain cases.
+    if (!hasPlaceholder(this.element_)) {
+      this.loadingContainer_.classList.add('i-amphtml-default-placeholder');
+    }
+  }
+
+  /**
+   * Whether the element is an Ad.
+   * @private
+   * @return {boolean}
+   */
+  isAd_() {
+    // Not Implemented
+    return false;
+  }
+
+  /**
+   * Whether the element is small.
+   * Small elements get a different loader with does not have a logo and is
+   * just a spinner.
+   * @private
+   * @return {boolean}
+   */
+  isSmall_() {
+    const box = this.element_.getLayoutBox();
+    return !isTiny(this.element_) && (box.width <= 100 || box.height <= 100);
+  }
+
+  /**
+   * Some components such as Instagram require larger spinner due to
+   * branding guidelines.
+   * @private
+   * @return {boolean}
+   */
+  requiresLargeSpinner_() {
+    // Not Implemented
+    return false;
+  }
+
+  /**
+   * Whether element has an existing image placeholder.
+   * @private
+   * @return {boolean}
+   */
+  hasImagePlaceholder_() {
+    // Not Implemented
+    return false;
+  }
+}
+
+/**
+ * Whether the new loader experiment is enabled.
+ * @param {!Window} win
+ * @return {boolean}
+ */
+export function isNewLoaderExperimentEnabled(win) {
+  return isExperimentOn(win, 'new-loaders');
+}
+
+/**
+ * Whether the element is eligible for loaders based on new loader heuristics
+ * This is is called by `isLoadingEnabled_` in `custom-element.js`
+ * and is additional heuristics to what `isLoadingEnabled_` checks.
+ * @param {!AmpElement} element
+ * @return {boolean}
+ */
+export function isLoaderIneligible(element) {
+  const result = isTiny(element) || isImageWithPlaceholder(element);
+  return result;
+}
+
+/**
+ * Very small layout are not eligible for new loaders.
+ * @param {!AmpElement} element
+ * @return {boolean}
+ */
+function isTiny(element) {
+  const box = element.getLayoutBox();
+  return box.width < 50 || box.height < 50;
+}
+
+/**
+ * Whether element is an amp-img with an existing
+ * placeholder (e.g. blurry image placeholder)
+ * @param {!AmpElement} element
+ * @return {boolean}
+ */
+function isImageWithPlaceholder(element) {
+  return element.tagName === 'AMP-IMG' && hasPlaceholder(element);
```

Comment:
Do we mean to only opt-out elements with blurry placeholders? If so, this isn't exactly the blurry placeholder heuristic. You can check if the placeholder has CSS `.i-amphtml-blurry-placeholder`.

Reply:
Initially we were thinking any placeholder after talking to Andrew, for now I am going to only exclude our own blurry placeholder as you mentioned until we test in the wild and see if we want to to extend that.